web3_proxy/src/frontend/authorization.rs

```diff
@@ -482,12 +473,13 @@ impl Web3ProxyApp {
     pub async fn registered_user_semaphore(
         &self,
         authorization_checks: &AuthorizationChecks,
-    ) -> anyhow::Result<Option<OwnedSemaphorePermit>> {
+    ) -> Web3ProxyResult<Option<OwnedSemaphorePermit>> {
         if let Some(max_concurrent_requests) = authorization_checks.max_concurrent_requests {
             let user_id = authorization_checks
                 .user_id
                 .try_into()
-                .context("user ids should always be non-zero")?;
+                .or(Err(Web3ProxyError::UserIdZero))
+                .web3_context("user ids should always be non-zero")?;
```

Comment:
this should probably also be a bad request?

Reply:
i dont think its possible to get an error here unless there is a bug